### ISSUE TITLE
Prisma Migrate: Improve UX when switching providers

### ISSUE THREAD

**Issue by albertoperdomo** [prisma/prisma#4760]
## Problem

The current experience using Prisma Migrate Preview when switching providers is pretty bad. The change in database provider (e.g. SQLite to Postgres) is not detected. Prisma Migrate attempts to run these migrations against the shadow database of the new provider, leading to runtime or syntax errors. These errors don't make it easy for the developer to understand the root cause and the path to resolution (get rid of the old migrations and start a new migration history from scratch).

Example (SQLite to Postgres):

```
> DATABASE_URL="postgres://prismatest:prismatest@localhost:5432/prismatest" npx prisma migrate dev --preview-feature
Environment variables loaded from prisma/.env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "prismatest", schema "public" at "localhost:5432"

2020-12-08 17:22:57.423 WET [11205] ERROR:  relation "_prisma_migrations" does not exist at character 126
2020-12-08 17:22:57.423 WET [11205] STATEMENT:  SELECT "id", "checksum", "finished_at", "migration_name", "logs", "rolled_back_at", "started_at", "applied_steps_count" FROM "_prisma_migrations" ORDER BY "started_at" ASC
2020-12-08 17:22:57.832 WET [11207] ERROR:  syntax error at or near "AUTOINCREMENT" at character 76
2020-12-08 17:22:57.832 WET [11207] STATEMENT:  -- CreateTable
	CREATE TABLE "Post" (
	    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
	    "title" TEXT NOT NULL,
	    "content" TEXT,
	    "published" BOOLEAN NOT NULL DEFAULT false,
	    "authorId" INTEGER,
	    "tags" TEXT,
	    "slug" TEXT NOT NULL,

	    FOREIGN KEY ("authorId") REFERENCES "User"("id") ON DELETE SET NULL ON UPDATE CASCADE
	);
```

**Comment by do4gr** [comment]
Our approach to implementing this will be: Introduce a migrations_lock.toml file at the top level in the migrations folder. It's current content:
```toml 
# Please do not edit this file manually. 
provider = "postgresql"
```
This file will be created and maintained by the migration engine. It can be crosschecked against the provider in the prisma.schema to allow us to detect a providerswitch. This is something the engine should do on any CLI call. 

If the ME detects a switch a user facing error should be returned to the CLI which the CLI can then uses to prompt for permission to switch providers. At this time it will also inform the user what will happen when doing this:
- It will explain the limitations: all db properties of the old db which are not present in the prisma.schema will be lost (everything created using manual sql for example)
- the complete local migration history will be archived (folders and old schema_lock.toml moved to a subfolder)
- the migrations_lock.toml file will be updated with the new provider
- a new migration will be generated migrating an empty db of the new provider to the current state of the prisma.schema in one step
- the migration will be applied

Necessary implementation steps:
- [ ] Add creation of schema_lock.toml to all necessary ME RPC calls
- [ ] Add check for provider switch to all ME RPC calls
- [ ] Introduce new Provider Switch RPC call in ME
- [ ] Implement handling for Provider Switch User Facing Error for CLI that calls Provider Switch RPC after confirmation

**Comment by do4gr** [comment]
This is now implemented and will be released in 2.15

**Comment by nickreynke** [comment]
Thank you for the implementation. Was wondering where the file came from, since it wasn't directly mentioned in the release tag https://github.com/prisma/prisma/releases/tag/2.15.0.
